src/FSLibrary/MissionMixedImageNetworkSurvey.fs

```diff
@@ -0,0 +1,130 @@
+open StellarCorePeer
+
+let mixedImageNetworkSurvey (context: MissionContext) =
+    let oldNodeCount = 1
+    let newNodeCount = 2
+
+    // Set max survey phase durations to 3 minutes
+    let surveyPhaseDurationMinutes = 3
+
+    let newImage = context.image
+    let oldImage = GetOrDefault context.oldImage newImage
+
+    let oldName = "core-old"
+    let newName = "core-new"
+
+    // keypairs of old and new surveyed nodes
+    let oldSurveyedKeys = KeyPair.Random()
+    let newSurveyedKeys = KeyPair.Random()
+
+    // Allow 2/3 nodes consensus, so that one image version could fork the
+    // network in case of bugs.
+    let qSet =
+        CoreSetQuorumListWithThreshold(([ CoreSetName oldName; CoreSetName newName ], 51))
+
+    let oldCoreSet =
+        { name = CoreSetName oldName
+          keys = [| oldSurveyedKeys |]
+          live = true
+          options =
+              { CoreSetOptions.GetDefault oldImage with
+                    nodeCount = oldNodeCount
+                    accelerateTime = false
+                    surveyPhaseDuration = None
+                    quorumSet = qSet } }
+
+    let newCoreSet =
+        { name = CoreSetName newName
+          keys = [| KeyPair.Random(); newSurveyedKeys |]
+          live = true
+          options =
+              { CoreSetOptions.GetDefault newImage with
+                    nodeCount = newNodeCount
+                    accelerateTime = false
+                    surveyPhaseDuration = Some surveyPhaseDurationMinutes
+                    quorumSet = qSet } }
+
+    let coreSets = [ newCoreSet; oldCoreSet ]
+
+    context.Execute
+        coreSets
+        None
+        (fun (formation: StellarFormation) ->
+            formation.WaitUntilSynced coreSets
+
+            // Upgrade the network to the old protocol
+            let oldPeer = formation.NetworkCfg.GetPeer oldCoreSet 0
+            let oldVersion = oldPeer.GetSupportedProtocolVersion()
+            formation.UpgradeProtocol coreSets oldVersion
+
+            // Chose a new node to run the survey
+            let surveyor = formation.NetworkCfg.GetPeer newCoreSet 0
+
+            // Helper functions to run survey commands from `surveyor`
+            let startSurveyCollecting () =
+                let nonce = 42
+                LogInfo "startSurveyCollecting: %s" (surveyor.StartSurveyCollecting nonce)
+
+            let stopSurveyCollecting () = LogInfo "stopSurveyCollecting: %s" (surveyor.StopSurveyCollecting())
+
+            let surveyTimeSliceData (node: KeyPair) =
+                LogInfo "surveyTimeSliceData: %s" (surveyor.SurveyTimeSliceData node.AccountId 0 0)
+
+            let getSurveyResult () = LogInfo "getSurveyResult: %s" (surveyor.GetSurveyResult())
+
+            let waitSeconds (seconds: int) =
+                LogInfo "Waiting %i seconds" seconds
+                System.Threading.Thread.Sleep(seconds * 1000)
+
+            // Start survey collecting
+            startSurveyCollecting ()
```

Comment:
Should we add some checks for strings returned from http requests? For example, here it'd be "Requested network to start survey collecting". This helps verify that survey actually runs, since our HTTP server returns 200 even if core encounters an error.

Reply:
Good idea. I added some checks on HTTP responses in e13e973f3470dfc990a2b815956969165e83dab8.

Reply:
Thank you!